diff --git a/board.h b/board.h
index e49a0ed..bcff41e 100644
--- a/board.h
+++ b/board.h
@@ -10,7 +10,7 @@
 #define CONFIG_BOOT_SDCARD
 #define CONFIG_BOOT_MMC
 
-#define CONFIG_FATFS_CACHE_SIZE		 16 * 1024 // (unit: 512B sectors) needs to be bigger than both DTB + kernel
+//#define CONFIG_FATFS_CACHE_SIZE		 16 * 1024 // (unit: 512B sectors) needs to be bigger than both DTB + kernel
 #define CONFIG_SDMMC_SPEED_TEST_SIZE 1024 // (unit: 512B sectors)
 
 #define CONFIG_CPU_FREQ 1200000000
